### PR TITLE
support show create table in plan2 and tae

### PR DESCRIPTION
**Pull request by jianwan0214**
**What type of PR is this?**

- [ ] API-change
- [x] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

**Which issue(s) this PR fixes:**

issue #2653 

**What this PR does / why we need it:**

Not Available

**Special notes for your reviewer:**

Not Available

**Additional documentation (e.g. design docs, usage docs, etc.):**

Not Available

### CODE REVIEW

File: pkg/frontend/mysql_cmd_executor.go [hunk 1]
```diff
@@ -191,6 +193,96 @@ func (o *outputQueue) flush() error {
...
+	//get database
+	storage := ses.GetStorage()
+	txnHandler := ses.GetTxnHandler()
+	db, err := storage.Database(dbName, txnHandler.GetTxn().GetCtx())
+	if err != nil {
+		return nil, err
+	}
+
+	//get table
+	table, err := db.Relation(string(tableName), txnHandler.GetTxn().GetCtx())
```
Comment (daviszhen) [inline pkg/frontend/mysql_cmd_executor.go:217]: You do not need read the attributes again. The rewritten sql has been executed and done for you.

File: pkg/frontend/mysql_cmd_executor.go [hunk 2]
```diff
@@ -191,6 +193,96 @@ func (o *outputQueue) flush() error {
...
+			createStr += pkStr
+		}
+	}
+
+	if rowCount != 0 {
+		createStr += "\n"
+	}
+	createStr += ")"
+
+	b := &types.Bytes{}
```
Comment (daviszhen) [inline pkg/frontend/mysql_cmd_executor.go:270]: You donot need to make a new batch. Just extract the needed columns from the batch after the batch been transformed into the row style.

**Review by daviszhen — CHANGES_REQUESTED** [review]

File: pkg/frontend/mysql_cmd_executor.go [hunk 3]
```diff
@@ -1531,6 +1578,25 @@ func (cwft *TxnComputationWrapper) SetDatabaseName(db string) error {
...
 func (cwft *TxnComputationWrapper) GetColumns() ([]interface{}, error) {
 	var err error
+	switch cwft.GetAst().(type) {
+	case *tree.ShowCreateTable:
+		cols := []*plan2.ColDef{
+			{Name: "Table"},
+			{Name: "Create Table"},
+		}
+		columns := make([]interface{}, len(cols))
+		for i, col := range cols {
```
Comment (daviszhen) [inline pkg/frontend/mysql_cmd_executor.go:1588]: the code block here is duplicate with the one below starts from 1601. It is unnessary

File: pkg/frontend/mysql_cmd_executor.go [hunk 4]
```diff
@@ -191,6 +198,43 @@ func (o *outputQueue) flush() error {
...
+/*
+handle show create table in plan2 and tae
+*/
+func handleShowCreateTable2(mrs *MysqlResultSet) error {
+	tableName := string(mrs.Data[0][1].([]byte))
+	createStr := fmt.Sprintf("CREATE TABLE `%s` (", tableName)
+	rowCount := 0
+	for _, d := range mrs.Data {
+		nullOrNot := ""
+		if d[7].(int8) != 0 {
```
Comment (daviszhen) [inline pkg/frontend/mysql_cmd_executor.go:210]: Define some constants with readable names for the number `1`,`7`,etc.

File: pkg/frontend/mysql_cmd_executor.go [hunk 5]
```diff
@@ -191,6 +198,43 @@ func (o *outputQueue) flush() error {
 	return nil
 }
 
+/*
+handle show create table in plan2 and tae
+*/
+func handleShowCreateTable2(mrs *MysqlResultSet) error {
```
Comment (daviszhen) [inline pkg/frontend/mysql_cmd_executor.go:204]: I think it is wrong here. As a assumption, the count of the table has 3 fields. Then the `handleShowCreateTable2` will be called 3 times. The  mrs is wrong, I think.

File: pkg/frontend/mysql_cmd_executor.go [hunk 6]
```diff
@@ -181,6 +183,11 @@ func (o *outputQueue) flush() error {
 		}
 	} else {
 		//send group of row
+		if o.showCreateTable {
+			if err := handleShowCreateTable2(o.mrs); err != nil {
+				return err
+			}
+		}
 		if err := o.proto.SendResultSetTextBatchRowSpeedup(o.mrs, o.rowIdx); err != nil {
```
Comment (daviszhen) [inline pkg/frontend/mysql_cmd_executor.go:191]: If it is `show create table` , it is unnecessary to send row here.

**Review by daviszhen — CHANGES_REQUESTED** [review]

**Review by ouyuanning — APPROVED** [review]

**Pull request merged**